File: framework/ui/src/io/intino/alexandria/ui/displays/components/OpenDialog.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.events.Event;
import io.intino.alexandria.ui.displays.events.actionable.OpenListener;
import io.intino.alexandria.ui.displays.notifiers.OpenDialogNotifier;

public class OpenDialog<DN extends OpenDialogNotifier, B extends Box> extends AbstractOpenDialog<DN, B> {
	private BaseDialog dialog;
	private OpenListener openListener = null;

	public OpenDialog(B box) {
        super(box);
    }

	public OpenDialog onOpen(OpenListener listener) {
		this.openListener = listener;
		return this;
	}

	public OpenDialog bindTo(BaseDialog dialog) {
    	this.dialog = dialog;
    	return this;
	}

	public void execute() {
		if (this.dialog == null) return;
		this.dialog.open();
		if (openListener != null) openListener.accept(new Event(this));
	}
}

File: framework/ui-elements/src/displays/components/OpenDialog.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractOpenDialog from "../../../gen/displays/components/AbstractOpenDialog";
import OpenDialogNotifier from "../../../gen/displays/notifiers/OpenDialogNotifier";
import OpenDialogRequester from "../../../gen/displays/requesters/OpenDialogRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import Actionable from "./Actionable";

const styles = theme => ({});

class OpenDialog extends AbstractOpenDialog {

	constructor(props) {
		super(props);
		this.notifier = new OpenDialogNotifier(this);
		this.requester = new OpenDialogRequester(this);
	};

}

export default withStyles(Actionable.Styles, { withTheme: true })(OpenDialog);
DisplayFactory.register("OpenDialog", withStyles(Actionable.Styles, { withTheme: true })(OpenDialog));